worldedit-core/src/main/java/com/sk89q/worldedit/regions/Region.java

```diff
@@ -451,24 +465,48 @@ default IChunkSet processSet(IChunk chunk, IChunkGet get, IChunkSet set, boolean
             for (int layer = getMinimumY() >> 4; layer <= getMaximumY() >> 4; layer++) {
                 int by = layer << 4;
                 int ty = by + 15;
-                if (containsEntireCuboid(bx, tx, by, ty, bz, tz)) {
-                    set.setBlocks(layer, FaweCache.INSTANCE.EMPTY_CHAR_4096);
-                    processExtra = true;
-                    continue;
-                }
-                char[] arr = set.load(layer);
-                for (int y = 0, index = 0; y < 16; y++) {
-                    for (int z = 0; z < 16; z++) {
-                        for (int x = 0; x < 16; x++, index++) {
-                            if (arr[index] != 0 && contains(x, y, z)) {
-                                arr[index] = 0;
-                                processExtra = true;
+
+                if (!(set instanceof IntSetBlocks)) {
+                    if (containsEntireCuboid(bx, tx, by, ty, bz, tz)) {
+                        set.setBlocks(layer, FaweCache.INSTANCE.EMPTY_CHAR_4096);
+                        processExtra = true;
+                        continue;
+                    }
+                    char[] arr = set.loadChars(layer);
+                    for (int y = 0, index = 0; y < 16; y++) {
```

Comment:
Same like in the `for` below?

Reply:
Refactored duplicate code into private method with next commit.